runtime/vm/createramclass.cpp

```diff
@@ -3081,6 +3082,19 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 #endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
 
 		result->classFlags = classFlags;
+		if (J9ROMCLASS_IS_ARRAY(romClass)) {
```

Comment:
also, its better to put this code with the rest of the array specific code in line 2800